imagepuller/internal/service/verify_test.go

```diff
@@ -4,81 +4,232 @@
 package service
 
 import (
+	"context"
+	"errors"
 	"fmt"
-	"io"
 	"log/slog"
 	"net/http/httptest"
 	"testing"
 
-	"github.com/containers/storage"
+	"github.com/edgelesssys/contrast/imagepuller/internal/remote"
 	"github.com/edgelesssys/contrast/imagepuller/internal/test/registry"
+	gcr "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/types"
 	"github.com/opencontainers/go-digest"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
-type StubStore struct {
-	putLayerLayer digest.Digest
-	storage.Store
-}
+var zeroDigest = "0000000000000000000000000000000000000000000000000000000000000000"
 
-func (s *StubStore) PutLayer(_, _ string, _ []string, _ string, _ bool, _ *storage.LayerOptions, _ io.Reader) (*storage.Layer, int64, error) {
-	return &storage.Layer{CompressedDigest: s.putLayerLayer}, 0, nil
-}
+var (
+	errDescriptor        = errors.New("descriptor")
+	errRemoteImage       = errors.New("remote image")
+	errRemoteIndex       = errors.New("remote index")
+	errCompressed        = errors.New("compressed layer")
+	errObtainingLayers   = errors.New("remote index")
+	errObtainingManifest = errors.New("compressed layer")
+	errPutLayer          = errors.New("put layer")
+)
```

Comment:
nit: I don't get why you would need these in the test. Just use errors.New("my unique error") for each test case, then you know you got the error you injected, right?

Reply:
Yes, but I need them both in the `tc.WantErr` and the stubs.

Reply:
Hm, that'd work, yes. Will change.